java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -1885,4 +1959,34 @@ public CompletableFuture<String> watch(@NonNull String[] keys) {
     public CompletableFuture<String> unwatch() {
         return commandManager.submitNewCommand(UnWatch, new String[0], this::handleStringResponse);
     }
+
+    // Hack: convert all `byte[]` -> `GlideString`. Better doing it here in the Java realm
+    // rather than doing it in the Rust code using JNI calls (performance)
+    private Object convertByteArrayToGlideString(Object o) {
+        if (o == null) return o;
+
+        if (o instanceof byte[]) {
```

Comment:
Just to confirm that it works fine and doesn't match non-byte-arrays (e.g. `Map[]`). I know java `instanceof T[]` isn't trivial and may not work as expected.

Reply:
The reason I moved it to the top was because `isArray()` captured the case of `byte[]`. With my first attempt it was down the "if" chain.